jetty-server/src/main/java/org/eclipse/jetty/server/AsyncContentProducer.java

```diff
@@ -316,119 +316,144 @@ public boolean isReady()
     {
         if (LOG.isDebugEnabled())
             LOG.debug("nextTransformedContent {}", this);
-        if (_rawContent == null)
-        {
-            _rawContent = produceRawContent();
-            if (_rawContent == null)
-                return null;
-        }
 
-        if (_transformedContent != null && _transformedContent.isEmpty())
+        while (true)
         {
-            if (_transformedContent != _rawContent)
-                _transformedContent.succeeded();
-            if (LOG.isDebugEnabled())
-                LOG.debug("nulling depleted transformed content {}", this);
-            _transformedContent = null;
-        }
-
-        while (_transformedContent == null)
-        {
-            if (_interceptor != null)
-            {
-                if (LOG.isDebugEnabled())
-                    LOG.debug("using interceptor to transform raw content {}", this);
-                _transformedContent = intercept();
-            }
-            else
+            if (_transformedContent != null && (_transformedContent.isSpecial() || !_transformedContent.isEmpty()))
```

Comment:
2nd & 3rd expressions are the opposite of the expressions at 343.
I think this could be simplified via:

```java
if (_transformed != null) 
{
    if (special && !empty) 
    {
        ...
    }
    else 
    {
        // block at 344
    }
}
```

In this way it would be IMHO clearer and less expression repetitions.

Reply:
Ok, done.